### DISCUSSION TITLE
Documentation on Typescript bounded useStore missing selector reference

### DISCUSSION THREAD

**Discussion by ampossardt** [pmndrs/zustand#2693] [Bug report]
### Bug Description

I'm currently using the typescript documentation to build some specific stores out, and noticed that the function for creating a generic bounded useStore for vanilla stores doesn't copy the selector into the `useStore` call which results in the whole store being returned.

The first line of this code:
```
const createBoundedUseStore = ((store) => (selector) => useStore(store))
```
should be this:
```
const createBoundedUseStore = ((store) => (selector) => useStore(store, selector))
```

### Reproduction Link

[Documentation link](https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md)

**Comment by dai-shi** [comment]
Nice catch! Can you send a PR please?
